### PR TITLE
Vng/extended exceptions

### PR DESCRIPTION
**Pull request by crankycoder**
this adds support for overloading the built in MetlogClient methods like .exception() with plugins.  the primary usecase right now is replacing the simple exception() or error() methods with a richer sentry plugin to capture detailed stack traces.

Made a minor sphinx doc fix updating a reference to CLIENT_WRAPPER to the new CLIENT_HOLDER

Also added metlog_hostname and metlog_pid entries into all fields so that we can track where a log came from

### CODE REVIEW

File: metlog/client.py [hunk 1]
```diff
@@ -135,11 +137,14 @@ class MetlogClient(object):
...
 
-    def __init__(self, sender=None, logger='', severity=6,
+    def __init__(self, sender, logger, severity=6,
                  disabled_timers=None, filters=None):
         """
         :param sender: A sender object used for actual message delivery.
         :param logger: Default `logger` value for all sent messages.
+                       This should be your application name and should
+                       not be modified for different instances of
+                       metlog within your the scope of your app.
```
Comment (rafrombrc) [inline metlog/client.py]: Wording is maybe a tiny bit strong here. "This is commonly set to be the name of the current application, and isn't modified for different instances of metlog within the scope of the same application."

File: metlog/client.py [hunk 2]
```diff
@@ -194,15 +203,17 @@ def send_message(self, msg):
                 return
         self.sender.send_message(msg)
 
-    def add_method(self, name, method):
+    def add_method(self, name, method, override):
```
Comment (rafrombrc) [inline metlog/client.py]: I'd much rather have override default to False. No need to make people think about it in except in the very few cases where it actually matters.

File: metlog/client.py [hunk 3]
```diff
@@ -241,6 +252,13 @@ def metlog(self, type, timestamp=None, logger=None, severity=None,
         logger = logger if logger is not None else self.logger
         severity = severity if severity is not None else self.severity
         fields = fields if fields is not None else dict()
+
+        # Populate the pid and hostname into the fields dictionary
+        # with a metlog_ prefix so that we don't clobber any other
+        # data
+        fields['metlog_pid'] = self.pid
+        fields['metlog_hostname'] = self.hostname
+
```
Comment (rafrombrc) [inline metlog/client.py]: Putting these in the `fields` dictionary is nice b/c it's a less invasive change, but I'm wondering if these shouldn't maybe be promoted to top level message attributes. Thoughts?

**Comment by rafrombrc** [comment]
Sorry to lag on this review, slipped through my cracks. I made a few comments, but you'll notice that they're all bike-sheddy nits, great work overall, thanks! I'm leaning towards moving the `pid` and `hostname` fields up to the top level of the message definition (instead of being in the otherwise free-form `fields` data structure), but am willing to be convinced otherwise if you think it's not worth it.

**Comment by crankycoder** [comment]
i've made those changes.  i'm mostly indifferent to where the hostname/pid are stored as far as top-level of fields as they are new attributes and we don't  have any backwards compatibility to deal with.

**Pull request merged**